includes/class-newspack-popups-model.php

```diff
@@ -259,10 +259,9 @@ public static function retrieve_popup_by_id( $post_id, $include_drafts = false )
 			'post_type'      => Newspack_Popups::NEWSPACK_POPUPS_CPT,
 			'posts_per_page' => 1,
 			'p'              => $post_id,
+			'post_status'    => $include_drafts ? [ 'draft', 'pending', 'future', 'publish' ] : 'publish',
```

Comment:
@adekbadek Can you help me having a look here?

This change breaks one test, but I wonder if the problem is in the test or in the logic here.

The problem I was trying to solve is that if I call `retrieve_popup_by_id` with `$include_drafts` as `true`, drafts are not being returned!

I added a `test_retrieve_popup_by_id` in the `test-model.php` file that passes with my change and fails if you try to revert the changes in this method.

Reply:
The code on `master` relies on the [dynamic default set for the query](https://developer.wordpress.org/reference/classes/wp_query/#status-parameters). In admin context, it will include the drafts, in non-admin context is will be just published posts. The argument name is misleading, should rather be called `$use_default_post_status` or something. For _actual_ draft-inclusion, there could be another argument, or a larger refactor that would make the code behave as suggested by the argument name.

Reply:
Gotcha, thanks for looking and clarifying, I was very confused to see it behaving differently.

I'll think of a solution

Reply:
And tests for the win!

Reply:
I have submitted a fix. Let me know what you think @adekbadek

Reply:
Looks good!